server/routerlicious/packages/services-client/src/restWrapper.ts

```diff
@@ -147,10 +153,26 @@ export class BasicRestWrapper extends RestWrapper {
                         this.request<T>(retryConfig, statusCode, false)
                             .then(resolve)
                             .catch(reject);
-                    } else if (error.response && error.response.status !== statusCode) {
-                        reject(error.response.status);
                     } else {
-                        reject(error);
+                        if (error?.response?.status) {
+                            reject(createFluidServiceNetworkError(error?.response?.status, error?.response?.data));
+                        } else if (error?.message === "Network Error") {
+                            // If a service is temporarily down or a browser resource limit is reached, Axios will throw
+                            // a network error with no status code (e.g. err:ERR_CONN_REFUSED or err:ERR_FAILED) and
+                            // error message, "Network Error". "Network Error" can be retried, and is not fatal.
+                            reject(new NetworkError(
+                                -1, /* code */
```

Comment:
If this bubbles up to another BasicRestWrapper, lets say Scribe calls Historian calls GitRest, and GitRest connection is down, Historian would send an `HTTP -1 "Network Error"` to Scribe's BasicRestWrapper.
1. Does Express.js allow setting status to -1?
2. When Scribe's BasicRestWrapper gets to line 158 and goes into `createFluidServiceNetworkError`, the switch case will result in `isFatal=true` and `canRetry=false`, but we would want to retry this type of error, right? I believe a quick fix would be to allow `isFatal` and `canRetry` variables to override their params in the default case of the switch/case

Reply:
Thanks for noticing that. I updated the error handling logic in BasicRestWrapper (based on this https://axios-http.com/docs/handling_errors). Now I don't use -1 anymore. Also, the new error handling setup will correctly take care of the network errors, which as we discussed offline, could surface through a different message (ENOTFOUND)